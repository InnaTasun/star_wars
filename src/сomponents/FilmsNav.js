import ElemByUrl from './ElemByUrl';
import styled from 'styled-components';

function FilmsNav(props) {
  const { dataArr, className } = props;
  let listItems = [];

  if (dataArr) {
    listItems = dataArr.map((film) => (
      <li className='films-list--film' key={film.episode_id}>
        <ElemByUrl url={film.url} mainProp='title' elemType='navImg' />
      </li>
    ));
  }

  return <ul className={`${className} films-list`}>{listItems}</ul>;
}

const StyledFilmsNav = styled(FilmsNav)`
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;

  .films-list {
    &--film {
      height: 290px;
      margin: 0.5rem;
      list-style-type: none;
      box-shadow: 7px 7px 7px rgba(0, 0, 0, 0.5);
    }
  }
`;

export default StyledFilmsNav;
