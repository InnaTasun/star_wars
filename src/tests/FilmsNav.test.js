import React from 'react';
import { render, screen } from '@testing-library/react';
import FilmsNav from '../сomponents/FilmsNav';

const data = [
  { title: 'Hero', episode_id: 1, url: 'https://domain/test/1/' },
  { title: 'SuperMan', episode_id: 2, url: 'https://domain/test/2/' },
  { title: 'SuperWoman', episode_id: 3, url: 'https://domain/test/3/' },
];

describe('FilmsNav component', () => {
  test('renders with data', () => {
    render(<FilmsNav dataArr={data} />);
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.queryByRole('list').childElementCount).toBe(3);
  });

  test('renders without data', () => {
    render(<FilmsNav />);
    expect(screen.getByRole('list').childElementCount).toBe(0);
  });

  test('renders with or without className', () => {
    let container = render(<FilmsNav className='main-class' />).container;
    let firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('main-class films-list');

    container = render(<FilmsNav className='test-class' />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('test-class films-list');

    container = render(<FilmsNav />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).not.toContain('main-class');
    expect(firstDiv.className).toContain('films-list');
  });
});

test("FilmsNav snapshot", () => {
  const { container } = render(<FilmsNav />);
  expect(container).toMatchSnapshot();
});