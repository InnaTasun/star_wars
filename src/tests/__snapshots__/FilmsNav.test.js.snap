// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsNav snapshot 1`] = `
<div>
  <ul
    class="sc-gsDKAQ enRQhp films-list"
  />
</div>
`;
